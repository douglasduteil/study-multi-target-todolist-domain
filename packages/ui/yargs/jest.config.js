//

module.exports = {
  preset: "ts-jest/presets/js-with-ts",
  testEnvironment: "node",
  collectCoverageFrom: ["src/**/*.{ts,tsx}"],
  transformIgnorePatterns: ["node_modules/(?!ramda)"],
  globals: {
    "ts-jest": {
      tsConfig: {
        strict: false,
        allowJs: true
      }
    }
  }
};
