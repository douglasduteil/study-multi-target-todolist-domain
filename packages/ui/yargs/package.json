{
  "name": "@todolist/ui.yargs",
  "description": "The Yargs todolist ui",
  "version": "1.0.0",
  "private": true,
  "main": "lib/index.js",
  "scripts": {
    "build": "tsc",
    "lint": "tslint -p tsconfig.json -t stylish",
    "test": "jest",
    "watch": "yarn build --watch --preserveWatchOutput"
  },
  "dependencies": {
    "@todolist/core": "^1.0.0",
    "dedent": "^0.7.0",
    "ramda": "^0.26.1",
    "s-ago": "^2.0.1",
    "tslib": "^1.9.3",
    "yargs": "^12.0.5"
  },
  "devDependencies": {
    "@types/dedent": "^0.7.0",
    "@types/jest": "^23.3.12",
    "@types/ramda": "^0.25.46",
    "@types/yargs": "^12.0.5",
    "jest": "^23.6.0",
    "ts-jest": "^23.10.5",
    "tslint": "^5.12.1",
    "typescript": "^3.2.2"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/douglasduteil/study-multi-target-todolist-domain.git"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "bugs": {
    "url": "https://github.com/douglasduteil/study-multi-target-todolist-domain/issues"
  },
  "homepage": "https://github.com/douglasduteil/study-multi-target-todolist-domain#readme"
}
