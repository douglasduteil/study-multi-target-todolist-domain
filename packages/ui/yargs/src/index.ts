//

import * as list from "./commands/list";

//

export { main } from "./main";

//

export const commands = {
  list
};
