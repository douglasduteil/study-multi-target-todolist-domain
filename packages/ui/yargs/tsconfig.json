{
  "extends": "../../../tsconfig.json",
  "compilerOptions": {
    "baseUrl": ".",
    "declaration": true,
    "outDir": "lib",
    "resolveJsonModule": true
  },
  "files": ["src/index.ts"]
}
