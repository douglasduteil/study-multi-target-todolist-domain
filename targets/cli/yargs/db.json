{
  "index": 0,
  "todos": []
}