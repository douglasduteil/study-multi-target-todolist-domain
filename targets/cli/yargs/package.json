{
  "name": "@todolist/target.yargs",
  "version": "0.0.0",
  "description": "",
  "private": true,
  "main": "index.js",
  "scripts": {
    "build": "ncc build ./src/index.ts -o dist",
    "lint": "tslint -p tsconfig.json -t stylish",
    "integration": "jest --config integration/jest.config.js",
    "test": "jest",
    "start": "ncc run ./src/index.ts",
    "watch": "yarn start --watch"
  },
  "keywords": [],
  "author": "Douglas Duteil <douglasduteil@gmail.com> (douglasduteil.github.io)",
  "license": "WTFPL",
  "dependencies": {
    "@todolist/core": "^1.0.0",
    "@todolist/databases.lowdb": "^1.0.0",
    "@todolist/ui.yargs": "^1.0.0",
    "lowdb": "^1.0.0"
  },
  "devDependencies": {
    "@types/jest": "^23.3.12",
    "@types/node": "^10.7.1",
    "@types/lowdb": "^1.0.6",
    "@zeit/ncc": "^0.8.1",
    "prettier": "^1.15.3",
    "ts-loader": "^5.3.1",
    "tslint": "^5.12.0",
    "typescript": "^3.2.2"
  }
}
