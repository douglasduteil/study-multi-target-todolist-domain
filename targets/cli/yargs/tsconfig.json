{
  "extends": "../../../tsconfig.json",
  "compilerOptions": {
    "resolveJsonModule": true
  },
  "files": ["src/index.ts"]
}
