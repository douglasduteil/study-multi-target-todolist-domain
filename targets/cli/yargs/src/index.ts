//

import { filterTodosFn } from "@todolist/core";
import { Database } from "@todolist/databases.lowdb";
import { commands, main } from "@todolist/ui.yargs";
import LowdbFileSyncStorageAdapter from "lowdb/adapters/FileSync";
import { version } from "../package.json";

//

const lowDbDataSource = new Database(
  new LowdbFileSyncStorageAdapter("db.json")
);

const { AddCommand, ListCommand } = commands;

main({ version })
  .command(new AddCommand(lowDbDataSource))
  .command(new ListCommand(filterTodosFn, console, lowDbDataSource))
  .parse(process.argv.slice(2));
