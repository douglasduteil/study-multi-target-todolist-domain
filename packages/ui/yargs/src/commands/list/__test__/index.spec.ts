//

import yargs from "yargs";
import { builder } from "..";

//

test("should add list options to cli", () => {
  const log = jest.fn();
  builder(yargs.usage("foo list <options>"))
    .wrap(80)
    .showHelp(log as any);
  expect(log.mock.calls[0]).toMatchSnapshot();
});
