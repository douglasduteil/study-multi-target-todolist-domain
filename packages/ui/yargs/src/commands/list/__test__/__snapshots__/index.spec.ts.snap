// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should add list options to cli 1`] = `
Array [
  "foo list <options>

List Options:
  --all, -a   Show done todos that are normally hidden[boolean] [default: false]
  --json      Show information as a JSON array        [boolean] [default: false]
  --long, -l  Show extended information               [boolean] [default: false]

Options:
  --help     Show help                                                 [boolean]
  --version  Show version number                                       [boolean]",
]
`;
